### PR TITLE
Updated Design Patterns And Development Strategies' Example Scripts

### PR DESCRIPTION
**Pull request by shbenzer**
### **User description**
Design patterns and development strategies' example scripts were outdated, so I updated them to reflect the new changes.

### Description
updated index.md in all languages with new code examples

### Motivation and Context
Github updated https://github.com/SeleniumHQ/selenium/issues/new, and since Design patterns and development strategies uses the page for its example scripts, those scripts needed to be updated.

### Types of changes

- [x] Change to the site (I have double-checked the Netlify deployment, and my changes look good)
- [x] Code example added (and I also added the example to all translated languages)
- [ ] Improved translation
- [ ] Added new translation (and I also added a notice to each document missing translation)

### Checklist

- [x] I have read the [**contributing**](https://selenium.dev/documentation/en/contributing/) document.
- [x] I have used [hugo](https://gohugo.io/) to render the site/docs locally and I am sure it works.

___

### **PR Type**
enhancement

___

### **Description**
- Updated the design patterns and development strategies example scripts to align with the new GitHub issues page.
- Replaced outdated methods `setSummary` and `enterDescription` with new methods for setting various issue form fields.
- Added new methods to handle additional fields in the issue form, such as `setTitle`, `setWhatHappened`, `setHowToReproduce`, `setLogOutput`, `setOperatingSystem`, `setSeleniumVersion`, `setBrowserVersion`, `setDriverVersion`, and `setUsingGrid`.
- Changed the selector for the submit button to a more specific CSS selector.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>design_strategies.en.md</strong><dd><code>Update issue form handling methods in English documentation</code></dd></summary>
<hr>

website_and_docs/content/documentation/test_practices/design_strategies.en.md

<li>Updated method names and parameters for issue form fields.<br> <li> Replaced <code>setSummary</code> and <code>enterDescription</code> with new methods.<br> <li> Changed the submit button selector.<br> <li> Added new methods for additional issue form fields.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1949/files#diff-ef76a0a38d2ead69e3cef4401392425717c886b9e7b638ede34872bf953aec22">+105/-14</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>design_strategies.ja.md</strong><dd><code>Update issue form handling methods in Japanese documentation</code></dd></summary>
<hr>

website_and_docs/content/documentation/test_practices/design_strategies.ja.md

<li>Updated method names and parameters for issue form fields.<br> <li> Replaced <code>setSummary</code> and <code>enterDescription</code> with new methods.<br> <li> Changed the submit button selector.<br> <li> Added new methods for additional issue form fields.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1949/files#diff-28d6e246bdacb4f79a7c028fdca7421b962de1c636f40d2f5c1ecbf31c7cdcb1">+104/-14</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>design_strategies.pt-br.md</strong><dd><code>Update issue form handling methods in Portuguese documentation</code></dd></summary>
<hr>

website_and_docs/content/documentation/test_practices/design_strategies.pt-br.md

<li>Updated method names and parameters for issue form fields.<br> <li> Replaced <code>setSummary</code> and <code>enterDescription</code> with new methods.<br> <li> Changed the submit button selector.<br> <li> Added new methods for additional issue form fields.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1949/files#diff-56162a0dfaf860cc543f32e8bb12cd55b4e264593d2156e1b006872cbf05b0c2">+104/-14</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>design_strategies.zh-cn.md</strong><dd><code>Update issue form handling methods in Chinese documentation</code></dd></summary>
<hr>

website_and_docs/content/documentation/test_practices/design_strategies.zh-cn.md

<li>Updated method names and parameters for issue form fields.<br> <li> Replaced <code>setSummary</code> and <code>enterDescription</code> with new methods.<br> <li> Changed the submit button selector.<br> <li> Added new methods for additional issue form fields.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1949/files#diff-ea1494cca27373682a4fd232bd005ccb7ae28f6cdf3e7e80304ceb1d6d368883">+104/-14</a></td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: website_and_docs/content/documentation/test_practices/design_strategies.en.md [hunk 1]
```diff
@@ -52,18 +52,58 @@ public class EditIssue {
     this.driver = driver;
   }
 
-  public void setSummary(String summary) {
-    WebElement field = driver.findElement(By.name("summary"));
-    clearAndType(field, summary);
+  public void getStarted() {
+    WebElement field = driver.findElement(By.xpath("//*[contains(text(), 'Get Started')]"))
```
Comment (harsha509) [inline website_and_docs/content/documentation/test_practices/design_strategies.en.md]: missing semicolon

File: website_and_docs/content/documentation/test_practices/design_strategies.en.md [hunk 2]
```diff
@@ -52,18 +52,58 @@ public class EditIssue {
...
+    field.click()
```
Comment (harsha509) [inline website_and_docs/content/documentation/test_practices/design_strategies.en.md]: missing semicolon

**Review by harsha509 — COMMENTED** [review]
HI @shbenzer,

Thank you for the PR.

I understand the need to update the locators as the current ones are outdated.

However, the additional changes seem to be related to `issue template` updates, which might not be necessary in this context. Including them could also make the code sample appear larger than required.

It would be great  if we could focus solely on updating the locators according to the template, while keeping the page objects unchanged.

Please let me know your thoughts!

Thanks,
Sri

**Comment by shbenzer** [comment]
I'm not 100% sure I understand what you mean.

Are you saying to instead just change the current functions to be relevant to the new page, but not add functions for everything on the new page, so that we can keep the same number of functions and the example doesn't change significantly in length?

**Comment by shbenzer** [comment]
@harsha509 would you mind expanding on your comments here?

**Comment by harsha509** [comment]
> I'm not 100% sure I understand what you mean.
> 
> Are you saying to instead just change the current functions to be relevant to the new page, but not add functions for everything on the new page, so that we can keep the same number of functions and the example doesn't change significantly in length?

Hi @shbenzer,

Yes, we can proceed by fixing the existing issue rather than adding new functions.

Additionally, there are a couple of things I do not understand. For example, the locator issue_form_operating-system is not present on the page [SeleniumHQ Issue Submission](https://github.com/SeleniumHQ/selenium/issues/new). Could you please clarify this?

Thank you.

**Comment by shbenzer** [comment]
> Yes, we can proceed by fixing the existing issue rather than adding new functions.

@harsha509 Gotcha

> Additionally, there are a couple of things I do not understand. For example, the locator issue_form_operating-system is not present on the page [SeleniumHQ Issue Submission](https://github.com/SeleniumHQ/selenium/issues/new). Could you please clarify this?

The old code is based on what we access by clicking "Get Started" next to Bug Report. Since not only has [SeleniumHQ Issue Submission](https://github.com/SeleniumHQ/selenium/issues/new) changed to let you pick which issue you're creating, the page we originally utilized has also changed to have more and different fields ("what happened", "what operating system are you using", "log output", etc.). The issue_form_operating-system locator is for the "Operating System: What host operating system are you using to run Selenium?" form field.

**Comment by harsha509** [comment]
> > Yes, we can proceed by fixing the existing issue rather than adding new functions.
> 
> @harsha509 Gotcha
> 
> > Additionally, there are a couple of things I do not understand. For example, the locator issue_form_operating-system is not present on the page [SeleniumHQ Issue Submission](https://github.com/SeleniumHQ/selenium/issues/new). Could you please clarify this?
> 
> The old code is based on what we access by clicking "Get Started" next to Bug Report. Since not only has [SeleniumHQ Issue Submission](https://github.com/SeleniumHQ/selenium/issues/new) changed to let you pick which issue you're creating, the page we originally utilized has also changed to have more and different fields ("what happened", "what operating system are you using", "log output", etc.). The issue_form_operating-system locator is for the "Operating System: What host operating system are you using to run Selenium?" form field.

Understood now!

The new additions are from different pages and has different url's

https://github.com/SeleniumHQ/selenium/issues
https://github.com/SeleniumHQ/selenium/issues/new/choose
https://github.com/SeleniumHQ/selenium/issues/new?assignees=&labels=I-defect%2Cneeds-triaging&projects=&template=bug-report.yml&title=%5B%F0%9F%90%9B+Bug%5D%3A+

Since this is just an example for the page object of a single page (https://github.com/SeleniumHQ/selenium/issues/new), I think changes related to this page should be enough.

**Comment by shbenzer** [comment]
> Since this is just an example for the page object of a single page (https://github.com/SeleniumHQ/selenium/issues/new), I think changes related to this page should be enough.

@harsha509 If that's the case, may I change the link from https://github.com/SeleniumHQ/selenium/issues/new to https://github.com/SeleniumHQ/selenium/issues/new?assignees=&labels=I-defect%2Cneeds-triaging&projects=&template=bug-report.yml&title=%5B%F0%9F%90%9B+Bug%5D%3A+?

Otherwise I think I'll have to make more changes than you're recommending.

**Comment by harsha509** [comment]
> https://github.com/SeleniumHQ/selenium/issues
> https://github.com/SeleniumHQ/selenium/issues/new/choose
> https://github.com/SeleniumHQ/selenium/issues/new?assignees=&labels=I-defect%2Cneeds-triaging&projects=&template=bug-report.yml&title=%5B%F0%9F%90%9B+Bug%5D%3A+





> > Since this is just an example for the page object of a single page (https://github.com/SeleniumHQ/selenium/issues/new), I think changes related to this page should be enough.
> 
> @harsha509 If that's the case, may I change the link from https://github.com/SeleniumHQ/selenium/issues/new to https://github.com/SeleniumHQ/selenium/issues/new?assignees=&labels=I-defect%2Cneeds-triaging&projects=&template=bug-report.yml&title=%5B%F0%9F%90%9B+Bug%5D%3A+?
> 
> Otherwise I think I'll have to make more changes than you're recommending.


I thought to suggest the same initially,
but , I think these should be treated as three separate pages, each with its own PageObject. It wouldn’t make sense to combine them into one class. So suggested the above.

**Comment by shbenzer** [comment]
Just so I'm clear - would that mean the following changes:

1. remove the setters (e.g. setSummary(), set Title(), etc.)
2. replace the setters with getters for each link on [New Issue](https://github.com/SeleniumHQ/selenium/issues/new/choose)

This way we would only need one page object and can make the least amount of changes to the page.

**Comment by harsha509** [comment]
> Just so I'm clear - would that mean the following changes:
> 
> 1. remove the setters (e.g. setSummary(), set Title(), etc.)
> 2. replace the setters with getters for each link on [New Issue](https://github.com/SeleniumHQ/selenium/issues/new/choose)
> 
> This way we would only need one page object and can make the least amount of changes to the page.

Yes, that works too. The main goal is to help users understand the Page Object Model.

**Comment by shbenzer** [comment]
Cool, I'll try to knock that out tonight or tomorrow if I can

**Comment by shbenzer** [comment]
@harsha509 So I ran into a problem and created a new solution... let me know what you think

In order to illustrate PageFactory's ability to locate elements with the same name/id of the field I had to shift the site to [Bug Report](https://github.com/SeleniumHQ/selenium/issues/new?assignees=&labels=I-defect%2Cneeds-triaging&projects=&template=bug-report.yml&title=%5B%F0%9F%90%9B+Bug%5D%3A+) so that the EditIssue class can still have the following:

> // By default the PageFactory will locate elements with the same name or id
> // as the field. Since the issue_title element has an id attribute of "issue_title"
> // we don't need any additional annotations.
> private WebElement issue_title;
> 
> // But we'd prefer a different name in our code than "issue_body", so we use the
> // FindBy annotation to tell the PageFactory how to locate the element.
> @FindBy(id = "issue_body") private WebElement body;

This is different than what we discussed. Since the only ids on the old page contained hyphens, I couldn't use them as variable names in java, plus they would have had to be the tabs to Pull Requests, Issues, etc. and wouldn't have really fit the example anyway.

Let me know what you think since this does still increase the size of the example a tad. However, the example focuses on a single website for the page object model like page objects should.

**Review by harsha509 — APPROVED** [review]
Thank you @shbenzer !

**Pull request merged**